services/web/server/tests/unit/isolated/test_utils_rate_limiting.py

```diff
@@ -22,35 +26,39 @@ async def get_ok_handler(_request: web.Request):
     return web.json_response({"value": 1})
 
 
-@pytest.mark.parametrize(
-    "requests_per_second",
-    [0.5 * MAX_REQUEST_RATE, MAX_REQUEST_RATE, 2 * MAX_REQUEST_RATE],
-)
-async def test_global_rate_limit_route(requests_per_second, aiohttp_client):
-    #
+@pytest.fixture
+def client(event_loop, aiohttp_client: Callable) -> TestClient:
     app = web.Application()
     app.router.add_get("/", get_ok_handler)
 
-    client = await aiohttp_client(app)
-    # ---
+    return event_loop.run_until_complete(aiohttp_client(app))
+
 
+def test_rate_limit_route_decorator():
     # decorated function keeps setup
     assert get_ok_handler.rate_limit_setup == (MAX_NUM_REQUESTS, MEASURE_INTERVAL)
 
+
+@pytest.mark.parametrize(
+    "requests_per_second",
+    [0.5 * MAX_REQUEST_RATE, MAX_REQUEST_RATE, 2 * MAX_REQUEST_RATE],
+)
+async def test_global_rate_limit_route(requests_per_second: float, client: TestClient):
+    # WARNING: this test has some timings and might fail when using breakpoints
+
     # Creates desired stream of requests for 1 second
-    TOTAL_TEST_TIME = 1  # secs
     num_requests = int(requests_per_second * TOTAL_TEST_TIME)
     time_between_requests = 1.0 / requests_per_second
 
-    futures = []
+    tasks = []
     t0 = time.time()
-    while len(futures) < num_requests:
+    while len(tasks) < num_requests:
         t1 = time.time()
-        futures.append(asyncio.create_task(client.get("/")))
+        tasks.append(asyncio.create_task(client.get("/")))
         time.sleep(time_between_requests - (time.time() - t1))
```

Comment:
> not sure I understand why we do not use an external rate limiter, but it was already there I guess.

Yes it was there already. This is just a fix and some cleanup

@sanderegg Is there a way to rate limit specific entrypoints in our reverse proxy? If so, perhaps we could create a maintenance issue and add it.

Reply:
> Question: why sync sleep instead of asyncio here?

It was already there ... it paces the rate in which it hits the entrypoint ... but on second thoughts, if the tasks does not execute, the ``time.sleep`` will block it and might cause flaky test . I will change it